components/Swaiper.tsx

```diff
@@ -1,36 +1,34 @@
-import React, { useRef, useState } from "react";
-import {
-  View,
-  ViewStyle,
-  Image,
-  ImageStyle,
-  Text,
-  TextStyle,
-} from "react-native";
+import React, { useEffect, useRef, useState } from "react";
+import { View, ViewStyle, Image, ImageStyle, TextStyle } from "react-native";
 import DeckSwiper from "react-native-deck-swiper";
 import { IconButton } from "./IconButton";
 import { SuperButton } from "./SuperButton";
+import { userList } from "../api/user.api";
+import { User } from "../types";
+import { useUser } from "../utils/user-hook";
+import { Text } from "react-native-paper";
 
 interface Props {
   esa?: string;
 }
 
 export const Swaiper = (props: Props) => {
   const swiperRef = useRef<DeckSwiper<any>>(null);
+  const curentUser = useUser();
+  const [users, setUsers] = useState<User[] | null>(null);
 
-  const data = [
-    { id: 1, image: require("../assets/imagines/download.png") },
-    { id: 2, image: require("../assets/imagines/essa.jpg") },
-    {
-      id: 3,
-      image: require("../assets/imagines/123436278_351097649321786_8786909236462977611_n(2).jpg"),
-    },
+  const fetchUserList = async () => {
+    const response = await userList();
+    console.log("User list", response);
+    setUsers(
+      response
+      // .filter((user) => curentUser.user.id !== user.id)
```

Comment:
You should really consider filtering out your own profile, otherwise you will allow for matching yourself.